cla_backend/libs/eligibility_calculator/cfe_civil/cfe_response.py

```diff
@@ -18,6 +18,23 @@ def employment_allowance(self):
     def gross_upper_threshold(self):
         return self._result_summary["gross_income"]["proceeding_types"][0]["upper_threshold"]
 
+    # The 'is_xxx_eligible checks are actually looking for reasons of ineligibility
+    # hence we check for not 'ineligible' rather than the other way round
```

Comment:
The name of the function "is_gross_eligible" is misleading then, in the case of "not_known_yet" or "not_calculated". How about `is_ineligible_on_gross_income_test`? Or `did_gross_income_test_fail`

On the language, I'm keen that we start calling GI/DI/Capital things "tests". And they are "pass / fail" rather than "eligible" / "ineligible".

Reply:
Actually maybe it's a bit surprising these values are inverse - True meaning fail.

How about tri-state? `did_gross_income_test_pass` and return True/False/None - the latter to mean `not_yet_known`/`not_calculated`?